final release/format_data.py

```diff
@@ -0,0 +1,208 @@
+
+    # Setting training and testing sets
+    if flip_train_test == False:
+        train_x = asu_fr_
+        train_y = asu_fr_lab_
+
+        test_x = leftover_fr
+        test_y = leftover_fr_lab
+
+
+    elif flip_train_test == True:
+        train_x = leftover_fr
+        train_y = leftover_fr_lab
+
+        test_x = asu_fr_
+        test_y = asu_fr_lab_
+
+    
+    # Defining rescaling function
+    def rescale_imgs(images,dim=dim):
+        size = images.shape[0]
+        imgs = np.zeros((size, dim, dim))
+
+        for i in range(size):
+            imgs[i, :, :] = resize(images[i, :, :], (dim, dim))
+
+        return np.expand_dims(imgs, axis=3)
+
+    
+    
+
+    #Rescaling Images
+    sys.stdout = open(os.devnull, "w") # don't show printed output
+    
+    train_x = rescale_imgs(train_x, dim)
+    test_x = rescale_imgs(test_x, dim)
+
+    sys.stdout = sys.__stdout__ # show printed output
+
+    # Convert class vectors to binary class matrices (for Keras Dense)
+    if conf_matrix == True:
+        train_y = np_utils.to_categorical(train_y, 2)
+        test_y = np_utils.to_categorical(test_y, 2)
+
+        
+    return train_x, train_y, test_x, test_y    
+
+
+import numpy as np
+from keras.preprocessing.image import ImageDataGenerator
+
+# Adding tiny Gaussian Noise to ensure there are no vanishing and exploding gradients
+def add_noise(image):
+        image += (10e-10)*np.random.randn(image.shape[0], image.shape[1], 1)
+        return image
+
+    
+def data_gen(rotation_range=180, zoom_range=0.2, shift_range = 0.0, flip=True):
+    
+    # Defining Data Generator
+    datagen = ImageDataGenerator(
+                rotation_range = rotation_range,
+                zoom_range = zoom_range,
+                width_shift_range = shift_range,
+                height_shift_range = shift_range,
+                horizontal_flip = flip,
+                vertical_flip = flip,
+                fill_mode = "nearest",
+                preprocessing_function=add_noise)
+    
+    return datagen
+
+
+def data_pregenerate(images, labels, datagen, batch_size, data_mult, seed):
+    samples = images.shape[0]
+
+    # the .flow() command below generates batches of randomly transformed images
+    # and saves the results to the `preview/` directory
+    gen = datagen.flow(images, labels, batch_size=batch_size, seed=seed)
+
+    pro_images = np.zeros((images.shape[0]*data_mult, images.shape[1], images.shape[2], 1))
+    pro_labels = np.zeros((labels.shape[0]*data_mult, 2))
+    
+    for e in range(1, data_mult+1):
+        batch = 1
+        b = batch_size
+        b_start = samples*(e-1)
+
+        for X_batch, Y_batch in gen:
+            if batch < (samples/b):
+                pro_images[b_start+b*(batch-1):b_start+batch*b, :, :, :] = X_batch
+                pro_labels[b_start+b*(batch-1):b_start+batch*b, :] = Y_batch
+                
+            elif batch == (samples/b):
```

Comment:
Will this ever happen? `batch` is an `int` and `samples / b` is a `float`.

Reply:
It has. This fixed it (Again, I wrote this a while back, so my memory is a little foggy).

Reply:
I'll add an int()